src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeOperator.scala

```diff
@@ -143,90 +141,79 @@ class TezosNodeOperator(node: TezosRPCInterface) extends LazyLogging {
     * @param network        Which Tezos network to go against
     * @param offset         How many previous blocks to get from the start
-    * @param startBlockHash The block from which to offset, using the hash provided or the head if none is provided.
+    * @param hash           The block from which to offset, using the hash provided or the head if none is provided.
     * @param followFork     If the predecessor of the minLevel block appears to be on a fork, also capture the blocks on the fork.
     * @return               Blocks
     */
-  def getBlocks(network: String, offset: Int, startBlockHash: Option[String], followFork: Boolean): Try[List[Block]] =
-    startBlockHash match {
-      case None =>
-        getBlockHead(network).flatMap(head => getBlocks(network, head.metadata.header.level - offset + 1, head.metadata.header.level, startBlockHash, followFork))
-      case Some(hash) =>
-        getBlock(network, hash).flatMap(block => getBlocks(network, block.metadata.header.level - offset + 1, block.metadata.header.level, startBlockHash, followFork))
-    }
+  def getBlocks(network: String, offset: Int, hash: String, followFork: Boolean): Try[List[Block]] =
+    getBlock(network, hash, 0).flatMap(block =>
+      getBlocks(
+        network,
+        block.metadata.header.level - offset + 1,
+        block.metadata.header.level,
+        hash,
+        followFork)
+    )
 
   /**
     * Gets the blocks using a specified rage
     * @param network        Which Tezos network to go against
     * @param minLevel       Minimum block level
     * @param maxLevel       Maximum block level
-    * @param startBlockHash If specified, start from the supplied block hash.
+    * @param hash           If specified, start from the supplied block hash.
     * @param followFork     If the predecessor of the minLevel block appears to be on a fork, also capture the blocks on the fork.
     * @return               Blocks
     */
-  def getBlocks(network: String, minLevel: Int, maxLevel: Int, startBlockHash: Option[String], followFork: Boolean): Try[List[Block]] =
-    startBlockHash match {
-      case None =>
-        getBlockHead(network).flatMap(head => Try {
-          processBlocks(network, head.metadata.hash, minLevel, maxLevel, followFork = followFork)
-        })
-      case Some(hash) =>
-        getBlock(network, hash).flatMap(block => Try {
-          processBlocks(network, block.metadata.hash, minLevel, maxLevel, followFork = followFork)
-        })
+  def getBlocks(network: String, minLevel: Int, maxLevel: Int, hash: String, followFork: Boolean): Try[List[Block]] = {
+    val blocksInRange = getBlock(network, hash, 0)
+      .flatMap(block => processBlocks(network, block.metadata.hash, minLevel, maxLevel))
+    // If followFork is true, create logic for that
+    val blocksFromFork = followFork match {
+      case false => Try{List[Block]()}
+      case true => Try{List[Block]()}
+    }
+    //Need to find good implementation of liftM2
```

Comment:
have a look at my comment and see if  `map2` fits

Reply:
Maybe! I feel like the pattern matching isn't too bad here though